easyDiffractionApp/Logic/PyQmlProxy.py

```diff
@@ -231,15 +241,30 @@ def __init__(self, parent=None):
         # Status info
         self.statusInfoChanged.connect(self._onStatusInfoChanged)
         self.currentCalculatorChanged.connect(self.statusInfoChanged)
+        self.currentCalculatorChanged.connect(self.undoRedoChanged)
         self.currentMinimizerChanged.connect(self.statusInfoChanged)
+        self.currentMinimizerChanged.connect(self.undoRedoChanged)
         self.currentMinimizerMethodChanged.connect(self.statusInfoChanged)
+        self.currentMinimizerMethodChanged.connect(self.undoRedoChanged)
```

Comment:
Rather than adding more slots would it not make sense to just call `undoRedoChanged` at the end of each relevant slot method? Are you sure the order of signals will be maintained? What if `undoRedoChanged` slot is executed before the previous `statusInfoChanged` slot and the undo stack gets out of sync?